### PR TITLE
Update to SpiderMonkey 66.

### PR DESCRIPTION
**Pull request by jdm**
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-mozjs/458)

### CODE REVIEW

**Comment by jdm** [comment]
Depends on https://github.com/servo/mozjs/pull/172.

**Comment by jdm** [comment]
Windows error:
```
  = note: lld-link: error: undefined symbol: JSID_VOID
          >>> referenced by C:\Users\appveyor\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\233235a\src\jsimpls.rs:55
          >>>               libmozjs_sys-82ed1c8753c8895b.rlib(mozjs_sys-82ed1c8753c8895b.mozjs_sys.c31j43od-cgu.1.rcgu.o):(_ZN9mozjs_sys7jsimpls96_$LT$impl$u20$core..default..Default$u20$for$u20$mozjs_sys..generated..root..JS..PropertyKey$GT$7default17h69604fc21c10c03eE)
          
error: aborting due to previous error
```

**Comment by jdm** [comment]
Linux/macOS debug and non-debugmozjs builds are succeeding. Windows builds are succeeding.

**Comment by jdm** [comment]
This is ready for review.

File: .travis.yml [hunk 1]
```diff
@@ -17,8 +17,13 @@ addons:
       - llvm-toolchain-xenial-5.0
     packages:
       - autoconf2.13
+      - gcc-6
       - g++-6
       - clang-5.0
+  homebrew:
+    packages:
+      - python
+      - python@2
```
Comment (asajeffrey) [inline .travis.yml]: Oh joy, two versions of python. :/

File: Cargo.toml [hunk 2]
```diff
@@ -51,4 +51,7 @@ lazy_static = "1"
 libc = "0.2"
 log = "0.4"
 num-traits = "0.2"
-mozjs_sys = "0.61.8"
+mozjs_sys = "0.66.0"
+
+[patch.crates-io]
+mozjs_sys = { git = "https://github.com/servo/mozjs", branch = "smup66" }
```
Comment (asajeffrey) [inline Cargo.toml]: Must remember to unpatch this before merging!

File: src/rust.rs [hunk 3]
```diff
@@ -313,22 +311,21 @@ impl Runtime {
...
+            (script.as_bytes().as_ptr(), script.len() as c_uint)
         };
         assert!(!ptr.is_null());
-        let _ac = JSAutoCompartment::new(self.cx(), glob.get());
+        let _ac = JSAutoRealm::new(self.cx(), glob.get());
         let options = CompileOptionsWrapper::new(self.cx(), filename_cstr.as_ptr(), line_num);
 
         unsafe {
-            if !Evaluate2(self.cx(), options.ptr, ptr as *const u16, len as size_t, rval.into()) {
+            if !EvaluateUtf8(self.cx(), options.ptr, ptr as *const _, len as size_t, rval.into()) {
```
Comment (asajeffrey) [inline src/rust.rs]: Hoorah! Evaluating a UTF-8 string without going via UTF-16!

**Pull request merged**